src/webportal/src/app/job-submission/components/data/data-component.jsx

```diff
@@ -46,8 +46,14 @@ function reducer(state, action) {
 }
 
 export const DataComponent = React.memo((props) => {
-  // add pylonAddress to .env for local debug
-  const hdfsHost = config.hasOwnProperty('pylonAddress') ? getHostNameFromUrl(config.pylonAddress) : window.location.hostname;
+  const envsubRegex = /^\${.*}$/;
+  let hdfsHost;
+  if (!config.pylonAddress || envsubRegex.test(config.pylonAddress)) {
```

Comment:
Here we use pylonAddress  to decide the hdfsHost address. Why not use WebHdfsUri directly.
Change the env config, if it contains WebHdfsUri, then use it. Otherwise use window.location.hostname.

.env file can be wrote like:
{
...
WEB_HDFS_URI=http://10.151.40.252:50700
}

Reply:
the env.js.template file does not contain WebHdfsUri key, unless we change the template file, which will change the window.ENV too

Reply:
Actually, I prefer change the template file and change the windows.ENV.
I will make code more reasonable and don't need to care about the logical between pylon and webhdfs
And this filed is optional.

If add a new filed in template file will cause any side-effect, please correct me.